### PR TITLE
Text3D Overlays and Entities fixes and no-AA

### PR DESCRIPTION
**Pull request by SamGondelman**
- text3D overlays set their shapeKeys correctly and update when their transparency changes
- text3D entities and overlays are not anti-aliased, which should make them clearer
- text3D entities and overlays have correct normals on their text which will improve lighting
- fixes a bunch of stencil buffer stuff to make it easier to use and more correct

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/ea8abaaf4504e314f6afa13acd112ae5/raw/745336eece19a6c220605ccff194b3e715b8c9d2/AATest2.js)
- In master you would see:
nonaatest8
- With this PR you'll see:
nonaatest9
(bottom left overlays have correctly colored text, background/text transparency is correct)
- Text overlays and entities should appear clearer
- Should fix [FB3488](https://highfidelity.fogbugz.com/f/cases/3488/Rendering-issues-on-text-entities-when-you-zoom-out)

### CODE REVIEW

**Comment by SamGondelman** [comment]
AA is not applied to text or web entities or overlays:
nonaatest7

**Comment by YuppyPlays** [comment]
Awaiting build for QA.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3244/HighFidelity-Beta-PR11345-a955381726dd72e3bcbdaf81501a6881ae09e7f8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3244/HighFidelity-Beta-PR11345-a955381726dd72e3bcbdaf81501a6881ae09e7f8.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:
With script:

Master:
pic1

PR: 
pic2

PR: (AA should not be applied to Text/Web entities)
pic3

Looks like a **Pass** to me, though if the pictures lead to anything otherwise do let me know.

File: libraries/render-utils/src/GeometryCache.cpp [hunk 1]
```diff
@@ -1907,9 +1910,9 @@ class SimpleProgramKey {
 
 
     SimpleProgramKey(bool textured = false, bool transparent = false, bool culled = true,
-        bool unlit = false, bool depthBias = false, bool fading = false) {
+        bool unlit = false, bool depthBias = false, bool fading = false, bool isAA = true) {
         _flags = (textured ? IS_TEXTURED : 0) | (transparent ? IS_TRANSPARENT : 0) | (culled ? IS_CULLED : 0) |
-            (unlit ? IS_UNLIT : 0) | (depthBias ? HAS_DEPTH_BIAS : 0) | (fading ? IS_FADING : 0);
+            (unlit ? IS_UNLIT : 0) | (depthBias ? HAS_DEPTH_BIAS : 0) | (fading ? IS_FADING : 0) | (isAA ? IS_AA : 0);
```
Comment (ZappoMan) [inline libraries/render-utils/src/GeometryCache.cpp]: `IS_AA` and `isAA` are confusing, and actually contrary to the coding standard. You should use `IS_ANTIALIASED` and `isAntiAliased`... AA could be confused with something else...

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3269/HighFidelity-Beta-PR11345-f68a588323c70a093c1135349d61f8dea0c63132.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3269/HighFidelity-Beta-PR11345-f68a588323c70a093c1135349d61f8dea0c63132.dmg

**Pull request merged**